### DISCUSSION TITLE
Regarding free trial settings

### DISCUSSION THREAD

**Discussion by PpsKet** [Glench/ExtPay#143] [Q&A]
I was able to implement the free trial function and perform email authentication to use it. 

However, when I uninstalled the extension once and reinstalled it, it became "That email has already been used." and I could not authenticate.

I have set a trial period, but can I make it available again during the period?

**Comment by takkbbl** [comment]
I think it would be great to have a richer user management system where I could delete accounts so they can reuse their email or block accounts that don't follow the terms of use.

**Comment by jeffsdevinfo** [comment]
Right, same problem here

> I was able to implement the free trial function and perform email authentication to use it.
> 
> However, when I uninstalled the extension once and reinstalled it, it became "That email has already been used." and I could not authenticate.
> 
> I have set a trial period, but can I make it available again during the period?

Right, I have the same issue.  During testing, if I register a trial user, uninstall the extension, then reinstall it, I am now unable to use that same email for trial login or regular login.  Thus rendering the email broken for use. Is it possible to alter the system to either allow relogin for trial (continuing on the previous register date) or regular even though the account has not payed yet?  It would seem that relogging into the trial would be the way to go because the regular login does not recognize the email.  Or is there a separate work flow for this? 

Trial Login:
 "That email has already been used."

Regular Login:
"Sorry, we don't have that email in our records. Please try another or contact the extension developer."

**Reply by connallykendrick** [reply]
BIG problem for me. Very annoying - the emails don't show up in the backend to delete/reset out of trial users on extpay website either.

**Comment by connallykendrick** [comment]
See [#108](https://github.com/Glench/ExtPay/issues/108)

**Reply by connallykendrick** [reply]
Similar issue was here [#128](https://github.com/Glench/ExtPay/issues/128)
Glen confirmed it's now fixed in #108

**Comment by Glench** [comment]
We just released a change so that your users can now log in to existing free trials. If you want to reset a user's trial for some reason, please just email me! So far almost no one has emailed me asking to reset for a specific user so I can only assume it's not a problem!

**Reply by connallykendrick** [reply]
I think a big part of this for us is the accessibility of being able to remove / blacklist (or whatever else) without bottleneck of having to email and wait (as I am right now).

In my current case, if I'm looking to repeatedly test the trial initiation out for functionality and to make changes for how things go during/afterwards, this is definitely slowing that down. I don't see this being as much of a need down the road, except for test users, revoking access later on if need be. 

I Am confused at what this is for if none of the trials show up here that I've done in dev mode? Would you explain? _deleted_

**Reply by Glench** [reply]
I looked in the database and this is accurate — that extension has no trials at all.

**Reply by connallykendrick** [reply]
When is a trial "registered"? - Once the extension is Outside of dev mode? 

I've signed up for trials with the extension loaded in by file to test, but it's not on the cws yet - is this why they are not showing in the trial users dashboard?

**Reply by Glench** [reply]
Are you sure? The database contradicts what you're saying. There are no users in or outside test mode that have a value for trialStartedAt.
